.gitlab-ci-common.yml

```diff
@@ -0,0 +1,52 @@
+# .gitlab-ci-common.yml
+#
+# This gitlab-ci file defines shell helper functions to be used by any Mender
+# pipeline. After including the file, you need to source the different snippets.
+#
+# Add it to the project in hand through Gitlab's include functionality:
+#
+# include:
+#   - project: 'Northern.tech/Mender/mendertesting'
+#     file: '.gitlab-ci-common.yml'
+#
+# And then source the desired functions in your job:
+#
+# my-fantastic-job:
+#   before_script:
+#     - *mender_ci_save_tmp_artifact
+#
+# Read the documentation for each function for more details
+#
+
+
+# Usage: mender_ci_save_tmp_artifact file-to-save
+#
+# Saves the given file in S3 bucket temporary storage.
+# For security, it computes the SHA256 checksum and saves it in
+# ${CI_PROJECT_DIR}/checksums. It is responsibility of the caller to save the
+# directory as part of the CI job artifacts:
+#   artifacts:
+#     paths:
+#       - checksums
+# Later jobs that will call mender_ci_load_tmp_artifact have to define "needs:"
+# or "depends:" to load the checksums directory
+.mender_ci_save_tmp_artifact: &mender_ci_save_tmp_artifact |
+  function mender_ci_save_tmp_artifact {
+    if [ -z "$1" ] then; echo Usage $0 file-to-save; exit 1; fi
+    user_file="$1"
+    mkdir -p ${CI_PROJECT_DIR}/checksums
+    sha256sum ${user_file} > ${CI_PROJECT_DIR}/checksums/$(basename ${user_file}).sha256
+    aws s3 cp ${user_file} s3://mender-gitlab-tmp-storage/$CI_PROJECT_NAME/$CI_PIPELINE_ID/$(basename ${user_file})
+  }
+
+# Usage: mender_ci_load_tmp_artifact file-to-load
+#
+# Loads the given file from S3 bucket temporary storage. It checks the SHA256
+# checksum (see mender_ci_save_tmp_artifact).
+.mender_ci_load_tmp_artifact: &mender_ci_load_tmp_artifact |
+  function mender_ci_load_tmp_artifact {
+    if [ -z "$1" ] then; echo Usage $0 file-to-load; exit 1; fi
+    user_file="$1"
+    aws s3 cp s3://mender-gitlab-tmp-storage/$CI_PROJECT_NAME/$CI_PIPELINE_ID/$(basename ${user_file}) ${user_file}
```

Comment:
I think the `PROJECT_NAME` and `PIPELINE_ID` need to also be parameters, as this will differ in (downstream) pipelines...

Reply:
These functions are meant to be used in the same pipeline. Mimic what today we have in mender-convert and I want to reuse in mender-dist-packages.

To cover the use case you are referring to (to share between pipelines) I would implement different functions, where we can either get PROJECT_NAME  and PIPELINE_ID as parameters or maybe just generate random dir and return the name to the caller or something. We can think about it when the time comes for implementing such case.

Reply:
IMO this is close enough to the share across pipelines functionality that it might be worth supporting that... apart from that I would still recommend explicit parameters instead of relying on global variables (also as a general programming practice).

Reply:
Okey, you convinced me. I'll rework :)